tests/disk/disk.go

```diff
@@ -118,59 +115,113 @@ func getLoopDevName() string {
 	return diskName
 }
 
+/*
 // RunCommandWithSudo runs a command with sudo permissions
 func RunCommandWithSudo(cmd string) error {
 	return RunCommand("sudo " + cmd)
 }
+*/
 
 // RunCommand runs a command on the host
-func RunCommand(cmd string) error {
+func RunCommand(cmd string) ([]byte, error) {
 	substring := strings.Fields(cmd)
 	name := substring[0]
 	args := substring[1:]
 	stdout, err := exec.Command(name, args...).CombinedOutput() // #nosec G204
 	if err != nil {
-		return fmt.Errorf("run failed, cmd={%s} error={%v} output={%v}\"", cmd, err, stdout)
+		return stdout, fmt.Errorf("run failed, cmd={%s}, error={%v}, output={%s}", cmd, err, string(stdout))
 	}
-	return err
+	return stdout, err
 }
 
-func (disk *Disk) CreateFileSystem(fstype string) error {
-	createMkfsCommand := "mkfs -t " + fstype + " " + disk.DiskPath
-	err := RunCommandWithSudo(createMkfsCommand)
+func (disk *Disk) CreateFilesystem(fstype string) error {
+	var mkfsCommand string
+	switch fstype {
+	case "ext4":
+		mkfsCommand = "mkfs.ext4 -O quota -E quotatype=prjquota " + disk.DiskPath
+	case "xfs":
+		mkfsCommand = "mkfs.xfs " + disk.DiskPath
+	default:
+		return fmt.Errorf("error creating mkfs command. " +
+			"Error: invalid filesystem type")
+	}
+
+	_, err := RunCommand(mkfsCommand)
 	if err != nil {
 		return fmt.Errorf("error creating fileystem on loop device. Error : %v", err)
 	}
 	return nil
 }
 
-func MkdirAll(path string) error {
-	createMkdirCommand := "mkdir -p " + path
-	err := RunCommandWithSudo(createMkdirCommand)
+func (disk *Disk) Wipefs() error {
+	// List partitions
+	fdiskCommand := "fdisk -o Device -l " + disk.DiskPath
+	fdiskStdOut, err := RunCommand(fdiskCommand)
+	if err != nil {
+		return fmt.Errorf("error listing disk partitions. Error: %v", err)
+	}
+
+	// Wipe partitions
+	stringsAndSubstrings := regexp.MustCompile("\n("+disk.DiskPath+".*)").
+		FindAllStringSubmatch(string(fdiskStdOut), -1)
+	for _, stringAndSubstrings := range stringsAndSubstrings {
+		//Remove string match (which includes newline)
+		// leave behind substring (without newline)
+		substrings := stringAndSubstrings[1:]
+
+		//This range will be always be of length 1
+		for _, substring := range substrings {
+			_, err := RunCommand("wipefs -fa " + substring)
+			if err != nil {
+				return fmt.Errorf("error removing filesystem "+
+					"from partition %s. Error: %v",
+					substring, err,
+				)
+			}
+		}
+	}
+
+	// Wipe disk
+	wipefsDeviceCommand := "wipefs -fa " + disk.DiskPath
+	_, err = RunCommand(wipefsDeviceCommand)
+	if err != nil {
+		return fmt.Errorf("error wiping disk %s. Error: %v",
+			disk.DiskPath, err,
+		)
+	}
+	return nil
+}
+
+func MkdirAll(paths ...string) error {
```

Comment:
Why cant the `os.MkdirAll()` be used ?

Reply:
It can... this is how it was done.... so I went ahead with it.
Moving to os.MkdirAll

Reply:
This was how it was done earlier, so went with it.
Moving to os.MkdirAll()